python/hail/tests/test_linalg.py

```diff
@@ -379,6 +379,40 @@ def test_slicing(self):
                         (slice(None, None), slice(None, None))]:
             self.assertTrue(np.array_equal(bm[indices].to_numpy(), nd[indices]))
 
+        self.assertRaises(ValueError, lambda: bm[0, ])
+
+        self.assertRaises(ValueError, lambda: bm[9, 0])
+        self.assertRaises(ValueError, lambda: bm[-9, 0])
+        self.assertRaises(ValueError, lambda: bm[0, 11])
+        self.assertRaises(ValueError, lambda: bm[0, -11])
+
+        self.assertRaises(ValueError, lambda: bm[::-1, 0])
+        self.assertRaises(ValueError, lambda: bm[0, ::-1])
+
+        self.assertRaises(ValueError, lambda: bm[:0, 0])
+        self.assertRaises(ValueError, lambda: bm[0, :0])
+
+        self.assertRaises(ValueError, lambda: bm[0:9, 0])
+        self.assertRaises(ValueError, lambda: bm[-9:, 0])
+        self.assertRaises(ValueError, lambda: bm[:-9, 0])
+
+        self.assertRaises(ValueError, lambda: bm[0, :11])
+        self.assertRaises(ValueError, lambda: bm[0, -11:])
+        self.assertRaises(ValueError, lambda: bm[0, :-11])
```

Comment:
I don't feel strongly about this, but it's a bit incongruous to not have these `assertRaises` in a for loop like the previous ones.

Reply:
I'd prefer to leave it because I think its easier to read through the edge cases this way, since I don't have to use ``slice(a,b,c)`` notation, and it's also faster to debug if one of them breaks (which happened a bunch as I was catching bugs).